Library/Homebrew/download_strategy.rb

```diff
@@ -463,15 +463,10 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
       url = url.sub(%r{^https?://#{GitHubPackages::URL_DOMAIN}/}o, "#{domain.chomp("/")}/")
     end
 
-    output, _, _status = curl_output(
-      "--location", "--silent", "--head", "--request", "GET", url.to_s,
-      timeout: timeout
-    )
-    parsed_output = parse_curl_output(output)
-
-    lines = output.to_s.lines.map(&:chomp)
+    parsed_output = curl_head(url.to_s, timeout: timeout)
+    parsed_headers = parsed_output.fetch(:responses).map { |r| r.fetch(:headers) }
```

Comment:
Just checking that we want this to go 💥 if `responses` is missing?

Reply:
Yes, `responses` should always be in this, even if empty.